ReSharper.FSharp/src/FSharp.Psi/src/CodeStyle/FSharpFormatterInfoProvider.cs

```diff
@@ -0,0 +1,245 @@
+using JetBrains.Application.Settings;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Impl.CodeStyle;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.CodeStyle
+{
+  [Language(typeof(FSharpLanguage))]
+  public class FSharpFormatterInfoProvider :
+    FormatterInfoProviderWithFluentApi<CodeFormattingContext, FSharpFormatSettingsKey>
+  {
+    public FSharpFormatterInfoProvider(ISettingsSchema settingsSchema) : base(settingsSchema)
+    {
+      lock (this)
+      {
+        Describe<FormattingRule>()
```

Comment:
It'd probably be neat to split the rules into several methods, like it's done for other languages. Maybe to have `Formatting` and `Indent` for now, and to regroup later if needed .